lib/std/os/windows.zig

```diff
@@ -937,25 +937,50 @@ pub fn DeleteFile(sub_path_w: []const u16, options: DeleteFileOptions) DeleteFil
         .DELETE_PENDING => return,
         else => return unexpectedStatus(rc),
     }
-    var file_dispo = FILE_DISPOSITION_INFORMATION{
-        .DeleteFile = TRUE,
-    };
-    rc = ntdll.NtSetInformationFile(
-        tmp_handle,
-        &io,
-        &file_dispo,
-        @sizeOf(FILE_DISPOSITION_INFORMATION),
-        .FileDispositionInformation,
-    );
-    CloseHandle(tmp_handle);
-    switch (rc) {
-        .SUCCESS => return,
-        .DIRECTORY_NOT_EMPTY => return error.DirNotEmpty,
-        .INVALID_PARAMETER => unreachable,
-        .CANNOT_DELETE => return error.AccessDenied,
-        .MEDIA_WRITE_PROTECTED => return error.AccessDenied,
-        .ACCESS_DENIED => return error.AccessDenied,
-        else => return unexpectedStatus(rc),
+    defer CloseHandle(tmp_handle);
+    if (comptime builtin.target.os.version_range.windows.min.isAtLeast(.win10_rs1)) {
```

Comment:
CI tells me it fails with `STATUS_NOT_SUPPORTED: 0xc00000bb`, see https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/596a1078-e883-4972-9bbc-49e60bebca55. Windows Server x86_64 and aarch64 non-Windows server failing implies to me the logic itself may be flawed.

WASI tells https://github.com/bytecodealliance/wasmtime/issues/582 `Windows 10 1607+ has the feature to delete files with "POSIX semantics".`., which is corresponding to redstone https://en.wikipedia.org/wiki/Windows_10_version_history.

However, I have no idea yet which server version corresponds to that and/or if version detection for servers works correctly and which server version we are using: https://github.com/actions/runner-images/blob/main/images/win/Windows2019-Readme.md and https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md

Next steps:
- [ ] check if both branches work locally on win10 x86_64
- [ ] check if both branches work locally on win11 x86_64
- [ ] figure out windows server <-> windows version + functionality mapping
- [ ] figure out how windows server version detection works + ensure its correct

Reply:
The `STATUS_NOT_SUPPORTED` error that we are seeing here is regardless of the windows version and unrelated to the server version, 2022 supports it and so is 2016 i believe. This is caused by having `FILE_DISPOSITION_ON_CLOSE` in the flags. 

`FILE_DISPOSITION_ON_CLOSE` is used to set/clear the `FILE_DELETE_ON_CLOSE` , but since we don't have that anymore in the flags, it's not going to work as the handle is already open without it. Microsoft docs does not say that unfortunately.

`FILE_DISPOSITION_POSIX_SEMANTICS`  already implies immediate removal on the close of the POSIX handle so i believe `FILE_DISPOSITION_ON_CLOSE` is not needed.

Reply:
Thanks alot @xEgoist for the quick help. That fixed it.